### ISSUE TITLE
Option --ignore-paths  broken in 2.12.0

### ISSUE THREAD

**Issue by mariof67** [PyCQA/pylint#5437]
### Bug description

I have a project that works in Mac OS  12.0.1. I upgraded pylint from version 2.11.1 to 2.12.0, and noticed that pylint is complaining about the specification of option `--ignore-paths`. Virtual environment uses python 3.9.8, and astroid 2.9.0. This issue maybe related to [5194](https://github.com/PyCQA/pylint/issues/5194)

My environment calls pylint on a per file basis on the project, and the following command works on pylint 2.11.1: `pylint --ignore-paths "sample,tools" sample/test_sample.py`. The file is ignored.

On pylint 2.12.0 and 2.12.1, the same command returns a cryptic error:

`Usage: pylint [options]`

`pylint: error: ignore-paths value ([re.compile('sample|sample'), re.compile('tools|tools')]) should be of type regexp_paths_csv`

I have tried many combinations trying to solve this issue, and they include:

`pylint --ignore-paths "sample" sample/test_sample.py`
`pylint --ignore-paths ".*/sample/.*$" sample/test_sample.py`
`pylint --ignore-paths "^.*$" sample/test_sample.py`
`pylint --ignore-paths ".*" sample/test_sample.py`

All the above commands fail. I used sample/test_sample.py for showing the commands, but it can be any filename.

Please advise about the right way of using this `--ignore-paths` option. Please also comment about the `regexp_paths_csv` format so that I can use it properly.

### Command used

```shell
pylint --ignore-paths "sample,tools" sample/test_sample.py
```

### Pylint output

```shell
Usage: pylint [options]

pylint: error: ignore-paths value ([re.compile('.*|.*')]) should be of type regexp_paths_csv
```

### Expected behavior

No output. Command should have an exit code of 0.

### Pylint version

```shell
pylint 2.12.0
astroid 2.9.0
Python 3.9.8 (main, Nov 10 2021, 09:21:22)
[Clang 13.0.0 (clang-1300.0.29.3)]
```

### OS / Environment

Mac OS 12.0.1 (Big Sur), iTerm2 3.4.12

### Additional dependencies

alabaster==0.7.12
astor==0.8.1
astroid==2.9.0
attrs==21.2.0
autopep8==1.6.0
Babel==2.9.1
backports.entry-points-selectable==1.1.0
bcrypt==3.2.0
bleach==4.1.0
build==0.7.0
certifi==2021.10.8
cffi==1.15.0
cfgv==3.3.1
charset-normalizer==2.0.7
colorama==0.4.4
coverage==6.2
cryptography==3.4.8
distlib==0.3.3
docutils==0.17.1
exrex==0.10.5
filelock==3.3.2
flake8==4.0.1
flake8-annotations==2.7.0
flake8-bugbear==21.11.28
flake8-builtins==1.5.3
flake8-comprehensions==3.7.0
flake8-docstrings==1.6.0
flake8-secure-coding-standard==1.2.1
flake8-simplify==0.14.2
future==0.18.2
httplib2==0.20.2
identify==2.3.3
idna==3.3
imagesize==1.2.0
importlib-metadata==4.8.1
isort==5.10.0
jedi==0.17.2
Jinja2==2.11.3
keyring==23.2.1
lazy-object-proxy==1.6.0
MarkupSafe==2.0.1
mccabe==0.6.1
mypy==0.910
mypy-extensions==0.4.3
nodeenv==1.6.0
packaging==21.2
paramiko==2.8.1
parso==0.7.1
pep517==0.12.0
pkginfo==1.7.1
plantuml==0.3.0
platformdirs==2.4.0
pluggy==1.0.0
pre-commit==2.15.0
pybadges==2.2.1
pycodestyle==2.8.0
pycparser==2.20
pydocstyle==6.1.1
pyenchant==3.2.2
pyflakes==2.4.0
Pygments==2.10.0
pylint==2.12.0
pyls-mypy==0.1.8
PyNaCl==1.4.0
pyparsing==2.4.7
pyroma==3.2
python-jsonrpc-server==0.4.0
python-language-server==0.36.2
python-lsp-jsonrpc==1.0.0
python-lsp-server==1.3.2
pytz==2021.3
PyYAML==6.0
readme-renderer==30.0
requests==2.26.0
requests-toolbelt==0.9.1
rfc3986==1.5.0
rope==0.22.0
six==1.16.0
snowballstemmer==2.1.0
Sphinx==4.3.1
sphinx-rtd-theme==1.0.0
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-confluencebuilder==1.7.0
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==2.0.0
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-plantuml==0.22
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.5
sphinxcontrib-spelling==7.2.1
toml==0.10.2
tomli==1.2.2
tqdm==4.62.3
twine==3.6.0
types-cryptography==3.3.9
types-enum34==1.1.1
types-ipaddress==1.0.1
types-paramiko==2.8.2
types-requests==2.26.1
typing-extensions==3.10.0.2
ujson==4.2.0
urllib3==1.26.7
virtualenv==20.10.0
webencodings==0.5.1
wrapt==1.13.3
yapf==0.31.0
zipp==3.6.0

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue

> pylint: error: ignore-paths value ([re.compile('.*|.*')]) should be of type regexp_paths_csv

I added the documentation label because this error is unclear

**Comment by DanielNoord** [comment]
Found the culprit. Expect a fix in `2.12` 😄

**Comment by mariof67** [comment]
Thank you very much for the quick response! I appreciate it.
